### PR TITLE
type fixes

### PR DESCRIPTION
**Pull request by kristojorg**
This PR just has a few type fixes that will allow CPW to enable strict type checking.

### CODE REVIEW

File: packages/opds-web-client/src/AuthPlugin.ts [hunk 1]
```diff
@@ -7,15 +7,15 @@ import { AuthMethod, AuthCredentials } from "./interfaces";
 /** Applications can implement this interface if they would like to support authentication
     methods other than basic auth. A list of `AuthPlugin`s can be passed as a prop to the
     `OPDSCatalog` component. */
-interface AuthPlugin {
+interface AuthPlugin<T extends AuthMethod = AuthMethod> {
```
Comment (EdwinGuzman) [inline packages/opds-web-client/src/AuthPlugin.ts:10]: Is the `= AuthMethod` similar to passing a default type if the caller just calls `AuthPlugin` without a type after?

**Review by EdwinGuzman — APPROVED** [review]

**Comment by EdwinGuzman** [comment]
Can you just update the changelog/version for a new npm version?

File: packages/opds-web-client/src/AuthPlugin.ts (hunk 1 above)
Comment (kristojorg) [reply packages/opds-web-client/src/AuthPlugin.ts:10]: Yep exactly

**Comment by kristojorg** [comment]
Okay did that!

**Comment by kristojorg** [comment]
Could you merge this when you have a minute if you don't mind?

**Review by EdwinGuzman — APPROVED** [review]

**Pull request merged**